File: Make_The_String_Great/Make_The_String_Great.cpp
class Solution {
public:
    string makeGood(string s) {
		string res;
        vector<char> v;
		for(int i=0;i<s.length();++i){
			if(v.size()&&abs(s[i]-v.back()==32))
				v.pop_back();
			else
				v.push_back(s[i]);
		}
		for(auto c:v)res+=c;
		return res;
    }
};

File: Solutions/Make_The_String_Great/Make_The_String_Great.java
/**
 * 
 */
package github.com.ErnestL1n;
import java.util.*;
/**
 * @author https://github.com/ErnestL1n
 *
 */
//Leetcode.1544
public class Make_The_String_Great {

	/**
	 * @param args
	 */
	public static String makeGood(String s) {
		Stack<Character> stack=new Stack<>();
		for(int i=0;i<s.length();++i) {
			if(!stack.isEmpty()&&Math.abs(s.charAt(i)-stack.peek())==32)
				stack.pop();
			else
				stack.push(s.charAt(i));
		}
		char[] res=new char[stack.size()];
		int index=stack.size()-1;
		while(!stack.isEmpty())
			res[index--]=stack.pop();
		return new String(res);
	}
	public static void main(String[] args) {

	}

}
